This patch will repeat the reminder notification every 2 minutes until you either dismiss or snooze the reminder.

11-15-09:
Updated for 1.3.1
Index: /usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
+++ /usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
@@ -98,6 +98,9 @@ var AppAssistant = Class.create({
 		else if (launchParams.alarmclose){
 			//console.error("$$$$$$$$ AppAssistant handleLaunch : alarmclose");
 			this.closeReminder(launchParams.alarmclose);
+		}
+		else if (launchParams.playalarmsound){
+			this.playAlarmSound();
 		} 
 		else if (launchParams.dayChange){
 			this.dayChange();
@@ -241,7 +244,14 @@ var AppAssistant = Class.create({
 		}
 		//console.error("$$$$$$$$ AppAssistant closeReminder:end");
 	},
-	
+
+	playAlarmSound: function() {
+		if (this.openReminderAlert) {
+			this.openReminderAlert.playAlarmSound();
+			this.openReminderAlert = null;
+		}
+	},
+													
 	scheduleNextReminder: function(eventId) {
 		//console.error("$$$$$$$$ AppAssistant scheduleNextReminder: for event:"+eventId);
 		new Mojo.Service.Request('palm://com.palm.calendar', {
Index: /usr/palm/applications/com.palm.app.calendar/app/controllers/reminder-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.calendar/app/controllers/reminder-assistant.js
+++ /usr/palm/applications/com.palm.app.calendar/app/controllers/reminder-assistant.js
@@ -57,6 +57,9 @@ var ReminderAssistant = Class.create({
 			}
 		}
 		//console.error("$$$$$$$$ Reminder-assistant cleanup:end:");
+		
+		//remove the Notification repeat task
+		this.removePlayAlarmSoundTask();
 	},
 
 	remindersUpdated: function() {
@@ -90,6 +93,9 @@ var ReminderAssistant = Class.create({
 		this.scheduleAutoCloseTask();
 		this.updateDisplay();
 		//console.error("$$$$$$$$ Reminder-assistant updateMostRecentReminder:end");
+		
+		//setup the notification repeat task
+		this.schedulePlayAlarmSoundTask();
 	},
 
 	updateDisplay: function() {
@@ -312,6 +318,14 @@ var ReminderAssistant = Class.create({
 		this.closeAlert();
 		//console.error("$$$$$$$$ Reminder-assistant autoCloseAlert: end " + this.reminder.id);
 	},
+	
+	playAlarmSound: function() {
+		//temporarily set the snooze to 0 will immediately re-trigger allert to get our attention.
+		this.dismissed = false;
+		this.DEFAULT_SNOOZE = 0;
+		this.closeAlert();
+	},
+										
 	specialCleanup:function(eventId){
 		//console.error("$$$$$$$$ Reminder-assistant specialCleanup: start:**********"+eventId );
 		//we need to do this so that cleanup snoozes the right reminder
@@ -360,6 +374,29 @@ var ReminderAssistant = Class.create({
 					                	});
 		//console.error("$$$$$$$$ Reminder-assistant removeAutoCloseTask: end " + this.reminder.id);
 	},
+
+	schedulePlayAlarmSoundTask: function() {
+		//retrigger reminder every two minutes until we acknowledge it.
+		//change the value in addMinutes to a value other than 2 if you prefer a different time span for the notification repeat.
+		//ultimatly it would be best to add this to the preferences page.
+		var playSoundTime = new Date().addMinutes(2);
+
+		this.controller.serviceRequest('palm://com.palm.taskScheduler', {
+			method: 'updateTask',
+			parameters: {uri: 'palm://com.palm.applicationManager/open',
+			arguments: {'id': 'com.palm.app.calendar',
+				'params': {'playalarmsound': 'true'}},
+			key: 'calendar-playalarmsound',
+			start: {date: playSoundTime.toUTCString()}}
+			});
+	},
+
+	removePlayAlarmSoundTask: function() {
+		this.controller.serviceRequest('palm://com.palm.taskScheduler', {
+			method: 'removeTask',
+			parameters: {key: 'calendar-playalarmsound'}
+			});
+	},
 	
   	replaceHTML: function(element, content) {
 		//console.error("$$$$$$$$ Reminder-assistant replaceHTML:start");
